src/dev/typescript/run_check_ts_projects_cli.ts

```diff
@@ -0,0 +1,89 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { resolve } from 'path';
+
+import execa from 'execa';
+
+import { run } from '../run';
+
+const REPO_ROOT = resolve(__dirname, '../../../');
+import { File } from '../file';
+import { PROJECTS } from './projects';
+
+export async function runCheckTsProjectsCli() {
+  run(
+    async ({ log }) => {
+      const files = await execa.stdout('git', ['ls-tree', '--name-only', '-r', 'HEAD'], {
+        cwd: REPO_ROOT,
+      });
+
+      const isNotInTsProject: File[] = [];
+      const isInMultipleTsProjects: File[] = [];
+
+      for (const lineRaw of files.split('\n')) {
+        const line = lineRaw.trim();
+
+        if (!line) {
+          continue;
+        }
+
+        const file = new File(resolve(REPO_ROOT, line));
+        if (!file.isTypescript() || file.isFixture()) {
```

Comment:
a possible conflict with https://github.com/elastic/kibana/pull/32148/files#diff-6e3fffc3af7f0f6bec3018ea1c6d42aeR32 
depending on merge order we need to update checks or find a common solution

Reply:
I think it's fine to categorize that file as a fixture in https://github.com/elastic/kibana/blob/master/src/dev/file.ts#L55-L57, which could be done in #32148 so the merge order doesn't matter